runtime/src/bank.rs

```diff
@@ -1806,17 +1834,44 @@ impl Bank {
         let mut partitions = vec![];
         if parent_epoch < current_epoch {
             if current_slot_index > 0 {
+                // Generate special partitions because there are skipped slots
+                // exactly at the epoch transition.
+
                 let parent_last_slot_index = self.get_slots_in_epoch(parent_epoch) - 1;
-                partitions.push(self.partition_from_slot_indexes(
+
+                // ... for parent epoch
+                partitions.push(self.partition_from_slot_indexes_with_gapped_epochs(
                     parent_slot_index,
                     parent_last_slot_index,
                     parent_epoch,
                 ));
+
+                // needed for test_rent_eager_across_epoch_with_gap_under_multi_epoch_cycle,
+                // which depends on OperatingMode::Stable
+                #[cfg(test)]
+                let should_enable = true;
```

Comment:
Can the scope of this be narrowed to be within the `OperatingMode::Stable` branch of the `#[cfg(not(test))]` case?

Reply:
done!